tests/unit/s2n_tls13_support_test.c

```diff
@@ -43,6 +44,37 @@ int main(int argc, char **argv)
         EXPECT_SUCCESS(s2n_connection_free(conn));
     }
 
+    /* Server does not parse new TLS 1.3 extensions by default */
+    {
+        uint8_t new_extensions[] = { TLS_EXTENSION_SUPPORTED_VERSIONS, TLS_EXTENSION_KEY_SHARE };
+
+        struct s2n_connection *server_conn;
+        EXPECT_NOT_NULL(server_conn = s2n_connection_new(S2N_SERVER));
+
+        struct s2n_stuffer extension_data;
+        EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&extension_data, 0));
+        EXPECT_SUCCESS(s2n_stuffer_write_str(&extension_data, "bad extension"));
+
+        uint8_t original_data_size = s2n_stuffer_data_available(&extension_data);
+
+        struct s2n_array *extensions = s2n_array_new(sizeof(struct s2n_client_hello_parsed_extension*));
+        for (int i=0; i < sizeof(new_extensions) / sizeof(uint8_t); i++) {
+            struct s2n_client_hello_parsed_extension *extension;
+            EXPECT_NOT_NULL(extension = s2n_array_add(extensions));
+
+            extension->extension = extension_data.blob;
+            extension->extension_type = new_extensions[i];
+        }
+
+        EXPECT_SUCCESS(s2n_client_extensions_recv(server_conn, extensions));
+        /* None of the extensions parsed any data */
+        EXPECT_EQUAL(original_data_size, s2n_stuffer_data_available(&extension_data));
```

Comment:
perhaps repeat the test when tls 1.3 flag is enabled to show that the reverse is true?